compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -7773,9 +7773,24 @@ public void visit(BLangTypeTestExpr typeTestExpr) {
         if (types.isValueType(expr.getBType())) {
             addConversionExprIfRequired(expr, symTable.anyType);
         }
-        typeTestExpr.expr = rewriteExpr(expr);
-        typeTestExpr.typeNode = rewrite(typeTestExpr.typeNode, env);
-        result = typeTestExpr;
+        if (!typeTestExpr.isNotIsExpr) {
+            typeTestExpr.expr = rewriteExpr(expr);
+            typeTestExpr.typeNode = rewrite(typeTestExpr.typeNode, env);
+            result = typeTestExpr;
+            return;
+        }
+        BLangTypeTestExpr bLangTypeTestExpr = (BLangTypeTestExpr) TreeBuilder.createTypeTestExpressionNode();
+        bLangTypeTestExpr.expr = typeTestExpr.expr;
+        bLangTypeTestExpr.typeNode = typeTestExpr.typeNode;
```

Comment:
Shall we use [org.wso2.ballerinalang.compiler.desugar.ASTBuilderUtil.#createTypeTestExpr](https://github.com/ballerina-platform/ballerina-lang/blob/58763ce56db611b14ee48db117f38b35c2011408/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ASTBuilderUtil.java#L230) instead?